pkg/resourcemanager/disk/client.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package disk
+
+import (
+	"context"
+	"strings"
+
+	compute "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-10-01/compute"
+	"github.com/Azure/azure-service-operator/pkg/resourcemanager/config"
+	"github.com/Azure/azure-service-operator/pkg/resourcemanager/iam"
+	"github.com/Azure/azure-service-operator/pkg/secrets"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type AzureDiskClient struct {
+	SecretClient secrets.SecretClient
+	Scheme       *runtime.Scheme
+}
+
+func NewAzureDiskClient(secretclient secrets.SecretClient, scheme *runtime.Scheme) *AzureDiskClient {
+	return &AzureDiskClient{
+		SecretClient: secretclient,
+		Scheme:       scheme,
+	}
+}
+
+func getDiskClient() compute.DisksClient {
+	computeClient := compute.NewDisksClientWithBaseURI(config.BaseURI(), config.SubscriptionID())
+	a, _ := iam.GetResourceManagementAuthorizer()
```

Comment:
You should use the error here and pass it out of methods calling it so the error isn't swallowed.